data/scripts.js

```diff
@@ -1353,7 +1358,7 @@ exports.BattleScripts = {
 				}
 
 				// Remove rejected moves from the move list.
-				if (rejected && movePool.length) {
+				if (rejected && movePool.length - availableHP + ((!availableHP || hasMove['hiddenpower'] && move.id !== 'hiddenpower') ? 0 : 1) >= 5 - moves.length) {
```

Comment:
It counts how many unique moves would remain if the move selected in the current iteration were rejected and checks whether that count is enough to fill the empty move slots that would result from such rejection.